provider/aws/processes.go

```diff
@@ -594,34 +594,54 @@ func (p *AWSProvider) describeInstance(id string) (*ec2.Instance, error) {
 	return res.Reservations[0].Instances[0], nil
 }
 
-func (p *AWSProvider) describeTask(arn string) (*ecs.Task, error) {
-	res, err := p.describeTasks(&ecs.DescribeTasksInput{
+func (p *AWSProvider) describeTaskInner(arn string) (*ecs.Task, error) {
+	input := &ecs.DescribeTasksInput{
 		Cluster: aws.String(p.Cluster),
 		Tasks:   []*string{aws.String(arn)},
-	})
+	}
+
+    res, err := p.describeTasks(input)
 	if err != nil {
 		return nil, err
 	}
 
 	// check the build cluster too
 	for _, f := range res.Failures {
 		if f.Reason != nil && *f.Reason == "MISSING" && p.BuildCluster != p.Cluster {
-			res, err = p.describeTasks(&ecs.DescribeTasksInput{
+			buildInput := &ecs.DescribeTasksInput{
```

Comment:
This can be combined into the describeTasks() call and doesn't need a variable assignment.